Tweak goldschlager to be more complex (gold, girls)

coiax opened the pull request:
:cl: coiax
tweak: Goldschlager will have the same effect of gold when drunk; it is
around 9% gold in SS13.
tweak: Goldschlager is now only really enjoyed by teenage girls, others
will think it's not bad, but not great.
/:cl:

Goldschlager contains real gold flakes in the real world, and in-game
it's made from 10u vodka and 1u gold, so it should have some effects of
gold.

The existing descriptions describe it being enjoyed by teenage girls, so
that is now mechanically enforced.

---

Having drinks that do nothing mechanically other than be a taste, sprite and description is boring, so I'm trying to add more metaphorical colour to various drinks.

coiax committed:
Tweak goldschlager to be more complex (gold, girls)

:cl: coiax
tweak: Goldschlager will have the same effect of gold when drunk; it is
around 9% gold in SS13.
tweak: Goldschlager is now only really enjoyed by teenage girls, other
will think it's not bad, but not great.
/:cl:

Goldschlager contains real gold flakes in the real world, and in-game
it's made from 10u vodka and 1u gold, so it should have some effects of
gold.

The existing descriptions describe it being enjoyed by teenage girls, so
that is now mechanically enforced.

Iamgoofball commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -476,13 +476,35 @@ All effects don't start immediately, but rather get worse over time; the rate is
+	if(ishuman(exposed_mob))
+		var/mob/living/carbon/human/human = exposed_mob
+		// tgstation13 does not endorse underage drinking. laws may vary by your jurisdiction.
+		if(human.age >= 13 && human.age <= 19 && human.gender == FEMALE)
```
the age literally cannot be below 19 in game, no point in doing anything other than == 19

interestingusernam3 replied:
Isn't the minimum 17?

Iamgoofball replied:
right, i forgot

cacogen replied:
So it should be 17 to 19 then.

coiax replied:
I'm not sure what we gain by specifying a more narrow range; and we definitely dilute the joke.

ExcessiveUseOfCobblestone replied:
Isn’t the min age a config? Should be fine to put whatever range in there

coiax replied:
Future proofing for whatever method of lowering someone's age that may exist. Also, what if podpeople have their age start at 0? What if we measure (gender-changed) Ian's age as rounds alive, and then he decided to booze it up?

itseasytosee commented:
There is clearly a chemical reaction going on due to the bubbles.  It is a new substance, not a mixture.

Iamgoofball commented:
@itseasytosee that's an oversight imho

Jared-Fogle commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -476,13 +476,35 @@ All effects don't start immediately, but rather get worse over time; the rate is
+	if(!(methods & INGEST))
+		return
+
+	// 1/11 is the amount of gold required to react in [/datum/chemical_reaction/goldschlager]
```
I don't want this magic number in here, since it expects nobody will ever touch the recipe for goldschlager.

Dynamically get this number at runtime by checking the required_reagents for goldschlager.. Caching it is fine.

Rohesie commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -476,13 +476,35 @@ All effects don't start immediately, but rather get worse over time; the rate is
+	var/convert_amount = trans_volume * (1 / 11)
+	A.reagents.remove_reagent(/datum/reagent/consumable/ethanol/goldschlager, convert_amount)
+	A.reagents.add_reagent(/datum/reagent/gold, convert_amount)
+	. = ..()
```
```suggestion
	return  ..()
```

Rohesie commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -476,13 +476,35 @@ All effects don't start immediately, but rather get worse over time; the rate is
 
+/datum/reagent/consumable/ethanol/goldschlager/expose_mob(mob/living/exposed_mob, methods=TOUCH, reac_volume)
+	// Reset quality if someone's sharing the bottle
+	quality = initial(quality)
```
Add a reagent-specific variable with the new quality if the conditions meet instead.
Something like
```
/// This is really popular with a certain demographic.
var/teenage_quality = DRINK_VERYGOOD
```

coiax replied:
But I already did this on line 490?

coiax committed:
Merge branch 'master' into goldschlager-complexity

coiax committed:
Merge branch 'master' into goldschlager-complexity

coiax committed:
Change reagent; ratio at runtime, quality var

- The conversion level of goldschlager into gold is determined from the
  recipe for goldschlager at runtime.
- The quality of goldschlager for teenager girls is now its own var.

Mothblocks reviewed (approved):
@Rohesie Good by me, merge it if it's good by you.

Rohesie reviewed (changes requested):
One standing change request.
Should be ready for merge after.

coiax commented:
@Rohesie I did the request you asked for already though

Rohesie commented:
> 
> 
> @Rohesie I did the request you asked for already though

My bad, misread it.
